lib/promoted/ruby/client.rb

```diff
@@ -73,7 +80,12 @@ def deliver args, headers={}
           delivery_request_builder.set_request_params(args)
 
           if perform_checks?
-            Promoted::Ruby::Client::Settings.check_that_log_ids_not_set!(args)
+            begin
+              Promoted::Ruby::Client::Settings.check_that_log_ids_not_set!(args)
```

Comment:
Side-note: I don't know how slow rescue blocks are in Ruby.  If they're slow, we can change check to return the error directly.

Not worth changing this in this PR.  Probably not worth changing until we have a benchmark in place.

Reply:
I figure this (and under perform_checks) mostly only comes up initially debugging SDK integrations in any event.

Reply:
Oh wait now I get your original meaning...most benchmarks only consider the scenario that's actually rescued; just establishing the block should be trivial.